rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/aws/AWSBakeHandler.groovy

```diff
@@ -52,16 +57,35 @@ public class AWSBakeHandler extends CloudProviderBakeHandler {
       bakeRequest = bakeRequest.copyWith(vm_type: awsBakeryDefaults.defaultVirtualizationType)
     }
 
+    def bakeKey
     // TODO(duftler): Work through definition of uniqueness.
     bakeRequest.with {
       def enhancedNetworkingSegment = bakeRequest.enhanced_networking ? 'enhancedNWEnabled' : 'enhancedNWDisabled'
 
       if (bakeRequest.base_ami) {
-        return "bake:$cloud_provider_type:$region:$vm_type:$base_os:$bakeRequest.base_ami:$enhancedNetworkingSegment:$package_name"
+        bakeKey = "bake:$cloud_provider_type:$region:$vm_type:$base_os:$bakeRequest.base_ami:$enhancedNetworkingSegment:"
+      } else {
+        bakeKey = "bake:$cloud_provider_type:$region:$vm_type:$base_os:$enhancedNetworkingSegment:"
+      }
+
+      def name
+      if (bakeRequest.ami_name) {
+        name = bakeRequest.ami_name
       } else {
-        return "bake:$cloud_provider_type:$region:$vm_type:$base_os:$enhancedNetworkingSegment:$package_name"
+        name = package_name
       }
+
+      def suffix
+      if (bakeRequest.ami_suffix) {
+        suffix = bakeRequest.ami_suffix
+      } else {
+        suffix = clock.millis()
```

Comment:
I know you'll be reworking this PR anyway, but wanted to mention that including the timestamp here will cause the bake key to not work as intended. If the bake key is used to determine if a bake being requested has already been requested, it has to be possible to generate the same bake key on subsequent calls.